src/containers/Application/Part0.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react'
+import { useLocation } from 'wouter'
+import VaccineInfo from '../../components/ApplicationForm/VaccineInfo'
+import NavigationButtons from '../../components/NavigationButtons'
+import VerticalProgressBar from '../../components/VerticalProgressBar'
+import { useAuth } from '../../utility/Auth'
+import { useHackerApplication } from '../../utility/HackerApplicationContext'
+import { checkForError, validateFormSection } from '../../utility/Validation'
+
+export default () => {
+  const { application, updateApplication, forceSave } = useHackerApplication()
+  const [, setLocation] = useLocation()
+  const { logout } = useAuth()
+  const [errors, setErrors] = useState({})
+  const [loading, setLoading] = useState(false)
+
+  const validate = change => {
+    const newErrors = validateFormSection(change, 'vaccineInfo')
+    setErrors({ ...errors, ...newErrors })
+    return { ...errors, ...newErrors }
+  }
+
+  const save = async () => {
+    setLoading(true)
+    await forceSave()
+    setLoading(false)
+  }
+
+  const updateVaccineInfo = change => {
+    validate(change)
+    console.log(application, 'this is the application object')
+
+    updateApplication({
+      vaccineInfo: {
+        ...change,
+      },
+    })
+  }
+
+  /**
+   * Saves and moves to next page
+   */
+  const handleNavigation = async href => {
+    await save()
+    if (href === '/application/part-1') {
+      const newErrors = validate(application.vaccineInfo)
+      if (checkForError(newErrors)) {
+        window.alert('You can only participate in nwHacks if you are double-vaccinated by then!')
+        return
+      }
+    }
+    setLocation(href)
+    window.scrollTo(0, 0)
+  }
+
+  const handleLogout = async () => {
+    await save()
+    logout()
+  }
```

Comment:
This was in `part-1` because there was no previous page of the application when the user is on the first part. But now that `part-0` is the first part, you would need to delete the `handleLogout` in `part-1` and replace `firstButtonText` with `firstButtonOnClick` in `Part1.js`:
```
<NavigationButtons
        firstButtonText="Back"
        firstButtonOnClick={() => handleNavigation('/application/part-0')}
... 
      />
```

Reply:
sounds good! I actually thought about this and I felt like it make sense for people to save and logout from part 1, changing tho!

Reply:
True true - I get what you mean and its great that you thought of it! Actually I think your approach might make more sense - the user doesn't need to go back to part-0 once they declare their vaccination status and we probably don't want them to go back anyways. could you revert back to your original change instead? hope its not too much work haha 😅